app/Actions/Album/Tracks.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\Actions\Album;
+
+use App\Models\Album;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+
+class Tracks extends Action
+{
+	/**
+	 * @param string $albumID the ID of the album
+	 * @param mixed  $value   the value to be set
+	 */
+	public function set(string $albumID, UploadedFile $value): void
+	{
+		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
+		if ($album->track_id != null) {
+			Storage::delete("tracks/$album->track_id.xml");
```

Comment:
Please store the whole relative path in the `album` and not only the id. This would a) be in line with how the size variants work, and b) it will allow to use other paths in the future if the `tracks/` part is not hard-coded, but part of the DB field.

The attribute is called `short_path` in the model `SizeVariant`, because this is the terminology used by Flysystem.

Moreover, I personally, don't like the PHP syntax which allows you to put variables into strings. Here it is somewhat cleat that the variable is `track_id` because the dot `.` works as a visual separator, but personally I would prefer something like `'tracks/' . $album->track_id . '.xml'`. It is more verbose, but IMHO is clearer.

But anyway it should rather be

```suggestion
	public function set(string $albumID, UploadedFile $value): void
	{
		$album = $this->albumFactory->findAlbumOrFail($albumID, false);
		if ($album->track_short_path != null) {
			Storage::delete($album->track_short_path);
```